### DISCUSSION TITLE
Translucency

### DISCUSSION THREAD

**Discussion by TyberiusPrime** [curv3d/curv#109] [Ideas]
Hey guys,

great work with curv, 
I've been toying with OpenSAD for a while (hey, I even have a wrapper for the python wrapper)
and I like what I see in curv so far very much.

And at least it's not 'three statements in a trenchcoat masquerading as a language' like OpenSCAD :).

Anyway,
one feature I've not found in the documentation is way to control transparence (or at least make some shapes 
temporarily translucent) like the OpenSCAD debug operator #.

Is there something like this?

**Comment by doug-moen** [comment]
There is no translucency in Curv right now. But I would like to have this feature at some point.

OpenSCAD has translucency in the preview (but not the render view), for 3D objects only, because it uses OpenCSG for 3D preview, and that is an OpenCSG feature that they can use with little effort. The Curv internals are quite different.

In Curv, I would want general support for translucency in 2D and 3D shapes, as part of the model. A shape would have a continuously variable translucency field that specifies a translucency or alpha value at each point in 2D or 3D space. We would have Porter-Duff compositing operators: a generalization of `union` that takes translucency into account.

One use case is exporting images with an alpha channel. Another use case is exporting 3D voxel files with an alpha channel. The latter is useful for 3D printing. Shapeways.com has a Stratasys J750 3D printer that is a voxel printer, and allows you to control the colour and translucency independently at each voxel. So you could print your translucent 3D models at Shapeways.
